src/manager/evm/manager.rs

```diff
@@ -33,21 +34,29 @@ pub type MiddlewareImpl = SignerMiddleware<Provider<Http>, Wallet<SigningKey>>;
 const SUBNET_MAJORITY_PERCENTAGE: u8 = 60;
 const TRANSACTION_RECEIPT_RETRIES: usize = 10;
 const SUBNET_NAME_MAX_LEN: usize = 32;
+const RANDOM_PRIVATE_KEY: [u8; 32] = [0; 32];
 
 // Create type bindings for the IPC Solidity contracts
 abigen!(Gateway, "contracts/Gateway.json");
 abigen!(SubnetContract, "contracts/SubnetActor.json");
 abigen!(SubnetRegistry, "contracts/SubnetRegistry.json");
 
-pub struct EthSubnetManager<M: Middleware> {
-    eth_client: Arc<M>,
-    gateway_contract: Gateway<Arc<M>>,
-    registry_contract: SubnetRegistry<Arc<M>>,
+pub struct EthSubnetManager {
+    keystore: Arc<RwLock<PersistentKeyStore<ethers::types::Address>>>,
+    on_chain_info: OnChainInfo,
+}
+
+/// Keep track of the on chain information for the subnet manager
+struct OnChainInfo {
```

Comment:
What about calling it `IPCInfo` or `IPCConfig`? 
```suggestion
struct IPCInfo {
```

Reply:
This is updated.